lib/healthcheck.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * create the healthcheck handler
+ *
+ * @param {Object} keystone - keystone instance
+ * @returns {function} - a healthcheck handler to check the health status of keystone(e.g. db connection)
+ */
+function createHealthCheckHandler(keystone){
+  return function(){
+    var readyState = keystone.mongoose.connection.readyState
+    // ERR_CONNECTING_TO_MONGO
+    if (readyState === 0 || readyState === 3) {
+      return Promise.reject(new Error('Mongoose has disconnected'))
+    }
+    // CONNECTING_TO_MONGO
+    if (readyState === 2) {
+    return Promise.reject(new Error('Mongoose is connecting'))
+    }
+    // CONNECTED_TO_MONGO
+    return Promise.resolve()
+  }
+}
+
+module.exports = {
```

Comment:
- you can use `module.exports = createHealthCheckHandler` to make require part cleaner
- for more detail about exports & module.exports, you can refer to this [stakeoverflow post](https://stackoverflow.com/questions/7137397/module-exports-vs-exports-in-node-js/17944431#17944431).

Reply:
I'm not sure which one is better.

```
require('./healthcheck')(keystone)
```
or
```
require('./healthcheck').createHealthCheckHandler(keystone)
```

Any thought?

Reply:
```
const healthCheckHandler = require('./healthcheck')(keystone)
'/healthcheck': healthCheckHandler
```
or
```
const healthcheck = require('./healthcheck')
'./healthcheck': healthcheck(keystone)
```
looks clear for me